python/pyarrow/_dataset.pyx

```diff
@@ -27,6 +27,7 @@ from pyarrow.lib cimport *
 from pyarrow.includes.libarrow_dataset cimport *
 from pyarrow.compat import frombytes, tobytes
 from pyarrow._fs cimport FileSystem, FileInfo, FileSelector
+from pyarrow._parquet import ParquetReader
```

Comment:
I am not yet very familiar with building issues, but, both dataset and parquet are optional modules when building pyarrow. So I don't think we can simply import from `pyarrow._parquet` here (maybe putting the import inline where it is needed is sufficient?)

Reply:
Currently parquet is a hard dependency of dataset, so even this should be fine until we bring dataset into core. I will inline the import